cmd/endorser/main.go

```diff
@@ -147,18 +145,21 @@ func parseDateOrDefault(date string, value time.Time) (time.Time, error) {
 	return time.Parse(layout, date)
 }
 
-func parseDigest(input string) (*digest, error) {
+func computeBinaryDigests(path string) (*intoto.DigestSet, error) {
 	// We expect the input to be of the ALG:VALUE form.
-	parts := strings.Split(input, ":")
-	if len(parts) != 2 {
-		return nil, fmt.Errorf("got %s, want ALG:VALUE format", input)
-	}
-	if !strings.EqualFold("sha256", parts[0]) && !strings.EqualFold("sha2-256", parts[0]) {
-		return nil, fmt.Errorf("unrecognized hash algorithm (%q), must be one of sha256 or sha2-256", parts[0])
+	bytes, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read bytes from path %q", path)
 	}
-	digest := digest{
-		alg:   parts[0],
-		value: parts[1],
+
+	sum256 := sha256.Sum256(bytes)
+	sum512 := sha512.Sum512(bytes)
+	sum384 := sha512.Sum384(bytes)
+
+	digestSet := intoto.DigestSet{
+		"sha256":   hex.EncodeToString(sum256[:]),
```

Comment:
Shouldn't this be "sha2-256" for consistency? Or does some other process still assume we will use "sha256"?

Reply:
I kept `sha256` for consistency with SLSA provenances, but the endorsement verification is under our control, and for that we can use `sha2-256`. Changed to `sha2-256`.